### PR TITLE
implementation ( Content Analytics Search):  #30231 Implement Search form and results.

### LINKED ISSUE #30231: CA search: Implement Search form and results.
### Parent Issue

#30178

### User Story
As a user, I want to implement a search form and results display for the Content Analytics portlet, where the input is a Monaco editor that accepts JSON queries, waits for the response, and displays the results in a read-only Monaco editor.

### Acceptance Criteria

- Implement a Monaco editor as the input for users to enter JSON queries.
- Ensure the editor can validate and format JSON correctly.
- Send the JSON query to the backend and wait for the response.
- Display the results in a read-only Monaco editor once the response is received.

### dotCMS Version
master

### Proposed Objective
_to_define_

### Proposed Priority
_to_define_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs

- The backend endpoint is available to process JSON queries and return results.
- Analytics infrastructure should be working in the environment.

### Quality Assurance Notes & Workarounds

- Add the portlet to  the menu to have access.

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Initial version of the Content Analytics search. 

### Screenshots

image

image

This pull request introduces a new analytics search feature and updates related components, services, and configurations to support it. The most important changes include adding a new service for analytics search, updating the routing module to include a new resolver, and modifying the component to handle analytics search queries and display results.

### New Analytics Search Feature:

* [`core-web/libs/data-access/src/lib/dot-analytics-search/dot-analytics-search.service.ts`](diffhunk://#diff-088e89fee35bf9a339b6fee96c3b70a9b3951e498b953582efe6d7a8bbb8c07dR1-R43): Added `DotAnalyticsSearchService` to perform analytics search operations with methods for making POST requests to specific endpoints.
* [`core-web/libs/data-access/src/lib/dot-analytics-search/dot-analytics-search.service.spec.ts`](diffhunk://#diff-468bba47163d237961e388bddf913b9a105879d9ae68dbc2e5d7c8b2342d8fddR1-R37): Added unit tests for `DotAnalyticsSearchService` to verify the POST requests and response handling.
* [`core-web/libs/dotcms-models/src/lib/dot-content-analytics.model.ts`](diffhunk://#diff-d296d98d7e78a0ba6f48c6706ba8aa9f358aa41b033b449e69ad6bfaefc4b2b7R1-R4): Introduced `AnalyticsQueryType` enum to differentiate between default and cube query types.

### Component Updates:

* [`core-web/libs/portlets/dot-analytics-search/portlet/src/lib/dot-analytics-search/dot-analytics-search.component.ts`](diffhunk://#diff-fa944beb3f94e677d995ccf0c2bde8e94332d5f9767887e1c71974d18e1e580eR1-R66): Updated `DotAnalyticsSearchComponent` to integrate with the new service, handle user queries, and display results using `ngx-monaco-editor`.
* [`core-web/libs/portlets/dot-analytics-search/portlet/src/lib/dot-analytics-search/dot-analytics-search.component.html`](diffhunk://#diff-f345130de7c17933ec8236c79bbf53077fa6d6d8d187cd8589f083796ff0f886L1-R27): Modified the template to include sections for query input and results display.
* [`core-web/libs/portlets/dot-analytics-search/portlet/src/lib/dot-analytics-search/dot-analytics-search.component.spec.ts`](diffhunk://#diff-2269342f0e34e2946cf219e6f852372d86e5883c9df8066e93f4167d755cae4aL1-R71): Added tests for `DotAnalyticsSearchComponent` to ensure proper initialization and query handling.

### Routing and Configuration:

* [`core-web/apps/dotcms-ui/src/app/app-routing.module.ts`](diffhunk://#diff-6c7f381c5033c96909090cb6d24c4507c999ed3ccec1ec3b74958152f21a572fR76-R81): Added `DotEnterpriseLicenseResolver` to the route configuration for `analytics-search` path and updated the route to use `MenuGuardService`.
* [`core-web/libs/portlets/dot-analytics-search/portlet/jest.config.ts`](diffhunk://#diff-15deaf36e0a2d3f5b2fdd099a1d0640c96b5e7c0f5ac08dbe37b127a2a96aa66L3-R7): Updated Jest configuration to reflect the new project structure and added `@happy-dom/jest-environment` for testing. [[1]](diffhunk://#diff-15deaf36e0a2d3f5b2fdd099a1d0640c96b5e7c0f5ac08dbe37b127a2a96aa66L3-R7) [[2]](diffhunk://#diff-15deaf36e0a2d3f5b2fdd099a1d0640c96b5e7c0f5ac08dbe37b127a2a96aa66L21-R23)

These changes collectively introduce a robust analytics search feature, ensuring it is well-integrated and thoroughly tested.

### CODE REVIEW

File: core-web/libs/portlets/dot-analytics-search/portlet/src/lib/dot-analytics-search/dot-analytics-search.component.ts [hunk 1]
```diff
@@ -1,11 +1,66 @@
...
+    ANALYTICS__RESULTS_MONACO_EDITOR_OPTIONS = ANALYTICS_RESULTS_MONACO_EDITOR_OPTIONS;
+
+    readonly store = inject(DotAnalyticsSearchStore);
+
+    queryEditor = '';
+
+    /**
+     * Computed property to get the results from the store and format them as a JSON string.
+     */
+    results = computed(() => {
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics-search/portlet/src/lib/dot-analytics-search/dot-analytics-search.component.ts:42]: Use the results$ for signals

File: core-web/libs/portlets/dot-analytics-search/portlet/jest.config.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
 /* eslint-disable */
 export default {
-    displayName: 'dot-analytics-search',
+    displayName: 'portlets-dot-analytics-search',
     preset: '../../../../jest.preset.js',
     setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
-    coverageDirectory: '../../../../coverage/libs/portlets/dot-analytics-search/portlet',
+    globals: {},
+    coverageDirectory: '../../../../../target/core-web-reports/',
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics-search/portlet/jest.config.ts:7]: I believe the coverageDirectory is already defined in the base Jest configuration, so this attribute is not necessary.

File: core-web/libs/portlets/dot-analytics-search/portlet/jest.config.ts [hunk 3]
```diff
@@ -18,5 +19,6 @@ export default {
         'jest-preset-angular/build/serializers/no-ng-attributes',
         'jest-preset-angular/build/serializers/ng-snapshot',
         'jest-preset-angular/build/serializers/html-comment'
-    ]
+    ],
+    testEnvironment: '@happy-dom/jest-environment'
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-analytics-search/portlet/jest.config.ts:23]: Amazing!

**Review by nicobytes — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-analytics-search/portlet/tsconfig.json [hunk 4]
```diff
@@ -1,6 +1,7 @@
 {
     "compilerOptions": {
         "target": "es2022",
+        "useDefineForClassFields": false,
```
Comment (oidacra) [inline core-web/libs/portlets/dot-analytics-search/portlet/tsconfig.json:4]: Is this necessary?

**Review by oidacra — APPROVED** [review]

File: core-web/libs/portlets/dot-analytics-search/portlet/tsconfig.json (hunk 4 above)
Comment (hmoreras) [reply core-web/libs/portlets/dot-analytics-search/portlet/tsconfig.json:4]: not really removed.

**Review by nicobytes — APPROVED** [review]

**Pull request merged**